misc/scripts/generate-code-scanning-query-list.py

```diff
@@ -0,0 +1,156 @@
+import subprocess
+import json
+import csv
+import sys
+import os
+
+"""
+This script collects CodeQL queries that are part of code scanning query packs
+and prints CSV data to stdout that describes which packs contain which queries.
+
+Errors are printed to stderr. This script requires that 'git' and 'codeql' commands
+are on the PATH. It'll try to automatically set the CodeQL search path correctly,
+as long as you run the script from one of the following locations:
+ - anywhere from within a clone of the CodeQL Git repo
+ - from the parent directory of a clone of the CodeQL Git repo (assuming 'codeql' 
+   and 'codeql-go' directories both exist)
+"""
+
+# Define which languages and query packs to consider
+languages = [ "cpp", "csharp", "go", "java", "javascript", "python"]
+packs = [ "code-scanning", "security-and-quality", "security-extended" ]
+
+# Define CodeQL search path so it'll find the CodeQL repositories:
+#  - anywhere in the current Git clone (including current working directory)
+#  - the 'codeql' subdirectory of the cwd
+#
+# (and assumes the codeql-go repo is in a similar location)
+codeql_search_path = "./codeql:./codeql-go:"   # will be extended further down
+
+
+def prefix_repo_nwo(filename):
+    """
+    Replaces an absolute path prefix with a GitHub repository name with owner (NWO).
+    This function relies on `git` being available.
+
+    For example:
+        /home/alice/git/ql/java/ql/src/MyQuery.ql  
+    becomes:
+        github/codeql/java/ql/src/MyQuery.ql
+     
+    If we can't detect a known NWO (e.g. github/codeql, github/codeql-go), the
+    path will be truncated to the root of the git repo:
+        ql/java/ql/src/MyQuery.ql
+    
+    If the filename is not part of a Git repo, the return value is the
+    same as the input value: the whole path.
+    """
+    dirname = os.path.dirname(filename)
+
+    git_toplevel_dir_subp = subprocess.run(
+        "git -C '%s' rev-parse --show-toplevel"  % dirname,
+        shell=True, capture_output=True, text=True
+    )
+
+    if git_toplevel_dir_subp.returncode != 0:
+        # Not a Git repo
+        return filename
+    
+    git_toplevel_dir = git_toplevel_dir_subp.stdout.strip()
+    
+    # Detect 'github/codeql' and 'github/codeql-go' repositories by SHA of first commit
+    first_sha = subprocess.run(
+        "git -C '%s' rev-list --max-parents=0 HEAD" % dirname,
+        shell=True, capture_output=True, text=True
+    ).stdout.strip()
+
+    if first_sha == "b55526aa58a5b500fe5d1be2c7edd09075711d09": prefix = "github/codeql"
+    elif first_sha == "d14eb855fc88e086b587f6c9695b59eb230c79e7": prefix = "github/codeql-go"
+    else: prefix = os.path.basename(git_toplevel_dir)
+
+    return os.path.join(prefix, filename[len(git_toplevel_dir)+1:])
```

Comment:
Looking at the output from the most recent Actions run, this works fine for `github/codeql-go`, but for `github/codeql` the prefix is only `codeql` (ie the directory name and not the repo name), so you seem to be incorrectly hitting the `else` case.

I suspect the reason is that `actions/checkout` by default only fetches up to depth 1, for speed. Hence looking up the first commit is not going to work. It works here for `codeql-go` because you did a manual clone instead of using `actions/checkout`.

Do you have to match against the first commit this way? I think there are simpler alternatives, e.g. looking at the output of `git remote --get-url origin` or `gh repo view | head -n 1`.

Reply:
I've indeed now switched to using `git remote -v` as a heuristic for finding the remote!